File: src/com/designpatterns/creational/factory/Circle.java
package com.designpatterns.creational.factory;

public class Circle implements IShape {

    public static final String TAG = "Circle";

    private final float radius;

    public Circle(float radius) {
        this.radius = radius;
    }

    @Override
    public float area() {
        return (float) (3.14 * radius * radius);
    }

    @Override
    public float perimeter() {
        return (float) (2 * 3.14 * radius);
    }

    @Override
    public void draw() {
        System.out.printf("Circle:: area:%f, perimeter:%f%n", area(), perimeter());
    }
}

File: src/com/designpatterns/creational/factory/kotlin/Circle.kt
package com.designpatterns.creational.factory.kotlin

class Circle(private val radius: Float) : IShape {

    override fun area(): Float {
        return (3.14 * radius * radius).toFloat()
    }

    override fun perimeter(): Float {
        return (2 * 3.14 * radius).toFloat()
    }

    override fun draw() {
        System.out.printf("Circle:: area:%f, perimeter:%f%n", area(), perimeter())
    }

    companion object {
        const val TAG = "Circle"
    }
}